common/djangoapps/edxmako/shortcuts.py

```diff
@@ -90,7 +90,10 @@ def marketing_link(name):
             if all([host_name and 'edge' in host_name, 'http' in link_map[name]]):
                 return link_map[name]
             else:
-                return reverse(link_map[name])
```

Comment:
All the values I see in the `MKTG_URL_LINK_MAP` dictionary are URLs, which we then try to reverse here. Shouldn't that _always_ produce an exception?

Reply:
We could always update to:
```
    marketing_link = reverse(link_map[name])
    set_custom_metric('temp_marketing_link_reverse', marketing_link)
    return marketing_link
```
We could also have the error condition write:
```
    set_custom_metric('temp_marketing_link_reverse_failed', link_map[name])
```
Anyway, it is simple to add some quick temporary custom metrics to see what is actually coming in if it would help.

Reply:
Yeah, I'd be very curious to see what we're getting there. Probably worth including the `host_name` as well.

Reply:
I can repro this on the login page by passing in a garbage Host header. I think the reverse always fails.